src/v/archival/ntp_archiver_service.cc

```diff
@@ -2636,33 +2636,29 @@ ss::future<> ntp_archiver::garbage_collect() {
         }
     }
 
-    size_t successful_deletes{0};
-    co_await ss::max_concurrent_for_each(
-      to_remove,
-      _concurrency,
-      [this, &successful_deletes](
-        const cloud_storage::partition_manifest::lw_segment_meta& meta) {
-          auto path = manifest().generate_segment_path(meta);
-          return ss::do_with(
-            std::move(path), [this, &successful_deletes](auto& path) {
-                return delete_segment(path).then(
-                  [this, &successful_deletes, &path](
-                    cloud_storage::upload_result res) {
-                      if (res == cloud_storage::upload_result::success) {
-                          ++successful_deletes;
-
-                          vlog(
-                            _rtclog.info,
-                            "Deleted segment from cloud storage: {}",
-                            path);
-                      }
-                  });
-            });
-      });
+    std::deque<cloud_storage_clients::object_key> objects_to_remove;
+    for (const auto& meta : to_remove) {
+        const auto path = manifest().generate_segment_path(meta);
+        vlog(_rtclog.info, "Deleting segment from cloud storage: {}", path);
+
+        objects_to_remove.emplace_back(path);
+        objects_to_remove.emplace_back(
+          cloud_storage::generate_remote_tx_path(path));
+        objects_to_remove.emplace_back(
+          cloud_storage::generate_index_path(path));
+    }
+
+    retry_chain_node fib(
+      _conf->manifest_upload_timeout * objects_to_remove.size(),
```

Comment:
This timeout would become too large when we delete say 100 segments. We'd be backing off exponentially for  50 minutes.

Reply:
This was the timeout prior to the change. It was 3x manifest_upload_timeout for each delete segment call. So I just copied it blindly. These timeouts are hard to set as not every backend has batch support. Even if they did, the batch size vary. So do their API performance. It gets even more complicated as the performance depends on pool sizes...

Maybe we should move these timeouts inside remote where we can make the decisions per backend? In follow up work (as this is current behaviour and not making it worse)?

Maybe at least clamp it to some maximum value for the time being?

Reply:
Yeah, I guess 3x was acceptable. I think clamping it to something reasonable is a great idea. Don't really have a great intuition here. Maybe 30 or 40 seconds?

Reply:
It was 3x per delete_segment call. And for 100 segments it would have been 3x100x timeout.

Reply:
Right, it was 3x per deletion, but since they were done concurrently (4x) it worked out as less in practice.